ledger-core/virtual/execute/execute.go

```diff
@@ -824,6 +824,19 @@ func (s *SMExecute) stepSendCallResult(ctx smachine.ExecutionContext) smachine.S
 
 func (s *SMExecute) stepFinishRequest(ctx smachine.ExecutionContext) smachine.StateUpdate {
 	if s.migrationHappened {
+		summarySharedKey := object.SummarySharedKey{PulseNumber: s.pulseSlot.PulseData().PulseNumber}
+		callSummaryAccessor, ok := object.GetSummarySMSharedAccessor(ctx, summarySharedKey)
+
+		if ok {
+			callSummaryAccessor.Prepare(func(sharedCallSummary *object.SharedCallSummary) {
+				ref := s.execution.Outgoing
+				requests, ok := sharedCallSummary.Requests.GetObjectsKnownRequests(ref)
+				if ok {
+					requests.GetList(s.execution.Isolation.Interference).Finish(ref, s.execution.Result)
+				}
+			}).TryUse(ctx)
+		}
+
```

Comment:
above block is not correct, it doesn't account situation when SM Object was not able to pass in data. what I mean is that request can reach this step after migration faster than object's finalizer.

also, this code doesn't check result of TryUse. Doesn't wait for shared and so on...

Reply:
Fixed